rust/tool/systemd/src/install.rs

```diff
@@ -438,15 +438,21 @@ fn force_install(from: &Path, to: &Path) -> Result<()> {
 pub fn append_initrd_secrets(
     append_initrd_secrets_path: &Path,
     initrd_path: &PathBuf,
+    generation_version: u64,
 ) -> Result<()> {
     let status = Command::new(append_initrd_secrets_path)
         .args(vec![initrd_path])
         .status()
         .context("Failed to append initrd secrets")?;
     if !status.success() {
         return Err(anyhow::anyhow!(
-            "Failed to append initrd secrets with args `{:?}`",
-            vec![append_initrd_secrets_path, initrd_path]
+            indoc::indoc! {"
+            Failed to append initrd secrets for generation {} with args `{:?}`
+            It may be that the secret is not at its expected location or that the `boot.initrd.secrets` option has been misused for that generation.
+            If that is the case, either fix the situation or delete the generation using `nix-env --delete-generations`.
```

Comment:
You could have a ready to copy'n'paste command here, if you include the generation number in the nix-env command invocation.